domain/src/test/java/org/oppia/domain/audio/CellularDialogControllerTest.kt

```diff
@@ -0,0 +1,203 @@
+import org.oppia.util.logging.GlobalLogLevel
+import org.oppia.util.logging.LogLevel
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+import kotlin.coroutines.EmptyCoroutineContext
+
+/** Tests for [UserAppHistoryController]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class CellularDialogControllerTest {
+  @Rule
+  @JvmField
+  val mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+  @Inject
+  lateinit var cellularDialogController: CellularDialogController
+
+  @Inject
+  @field:TestDispatcher
+  lateinit var testDispatcher: CoroutineDispatcher
+
+  @Inject
+  @field:TestBlockingDispatcher
+  lateinit var testBlockingDispatcher: TestCoroutineDispatcher
+
+  private val coroutineContext by lazy {
+    EmptyCoroutineContext + testDispatcher
+  }
+
+  @Mock
+  lateinit var mockCellularDataObserver: Observer<AsyncResult<CellularDataPreference>>
+
+  @Captor
+  lateinit var cellularDataResultCaptor: ArgumentCaptor<AsyncResult<CellularDataPreference>>
+
+  @Before
+  @ExperimentalCoroutinesApi
+  @ObsoleteCoroutinesApi
+  fun setUp() {
+    setUpTestApplicationComponent()
+    testBlockingDispatcher.pauseDispatcher()
+  }
+
+
+  private fun setUpTestApplicationComponent() {
+    DaggerCellularDialogControllerTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testController_providesInitialLiveData_indicatesNotHideDialog()
+      = runBlockingTest(coroutineContext) {
+    val cellularDataPreference = cellularDialogController.getCellularDataPreference()
+    cellularDataPreference.observeForever(mockCellularDataObserver)
+    testBlockingDispatcher.advanceUntilIdle()
+
+    verify(mockCellularDataObserver, atLeastOnce()).onChanged(cellularDataResultCaptor.capture())
+    assertThat(cellularDataResultCaptor.value.isSuccess()).isTrue()
+    assertThat(cellularDataResultCaptor.value.getOrThrow().hideDialog).isFalse()
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testController_setHidePrefTrue_providesLiveData_indicatesHideDialog()
+      = runBlockingTest(coroutineContext) {
+    val appHistory = cellularDialogController.getCellularDataPreference()
+
+    appHistory.observeForever(mockCellularDataObserver)
+    cellularDialogController.setHideDialogPreference(true)
+    testBlockingDispatcher.advanceUntilIdle()
+
+    verify(mockCellularDataObserver, atLeastOnce()).onChanged(cellularDataResultCaptor.capture())
+    assertThat(cellularDataResultCaptor.value.isSuccess()).isTrue()
+    assertThat(cellularDataResultCaptor.value.getOrThrow().hideDialog).isTrue()
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testController_setHidePrefFalse_providesLiveData_indicatesNotHideDialog()
```

Comment:
Suggest also adding a test that demonstrates a new controller & observer properly retrieves the on-disk state from a previously set preference. You can simulate a new controller by calling setUpTestApplicationComponent() a second time (and can verify via step debugging that the instance of the cellular dialog controller is actually a different object during the second half of the test). This verifies the "restart app" functionality.